### PR TITLE
Added Scipy implementation for simpson

### PR DESCRIPTION
**Pull request by Priyanka-43**
# Scipy for Simpson
  - This pull request contain Scipy implementation for simpson.

## Linting
  - Pylint is 10/10

### CODE REVIEW

**Comment by Priyanka-43** [comment]
Hi, i am unable to fix my pytest error. I tried a lot. Can anyone help me in this?

**Comment by zbpetersbuf** [comment]
Yes this is because you are not doing a pytest test function, this defining class and if __name__ == "__main__": is specifically what @laserlab said not to do a couple weeks ago in class, I suggest looking at other peoples test functions

**Comment by Priyanka-43** [comment]
@zbpetersbuf  previously i was doing the same like not to use  name == "main", But it was still showing error. I will do one commit more to see.

**Comment by zbpetersbuf** [comment]
also the class TestCalculusFunctions(unittest.TestCase):, this is not pytest, pytest does not recognize this as a test function please look at other peoples test functions

**Comment by Priyanka-43** [comment]
Ok sure.

**Comment by zbpetersbuf** [comment]
ill check in the morning to merge after an approved review

**Comment by Priyanka-43** [comment]
Hi, I have tried different codes to run. There is always error in  scipy. Not giving correct result. If i am running in jupyter NB, it is giving result.

**Comment by zbpetersbuf** [comment]
Result: 99.0000020612203
=========================== short test summary info ============================
FAILED test_calculus.py::test_scipy_simpson - AssertionError: Expected 2.0, but got 99.0000020612203
assert np.False_

maybe the best way to fix this is to trouble shoot on your end, by making sure before you use np.isclose is actually close on the range you give it before making it into a test_function

File: test_calculus.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
 """
 test_calculus.py
 """
+
+import numpy as np
 import calculus as calc
+from calculus import scipy_simpson
```
Comment (laserlab) [inline test_calculus.py:7]: DO not make a second import statement of the same module, which is not even used

**Review by laserlab — CHANGES_REQUESTED** [review]
- doubled import statement
- repo has to be synced

**Comment by Priyanka-43** [comment]
Can anyone help me to resolve my pytest error? I have done correct implementation.

**Comment by zbpetersbuf** [comment]
@Priyanka-43 lines 35 and 36 why are you printing f"x and f"y this would only be in debugging or when you are making the code, if its in your pull request you should know it works, have you tried removing these? perhaps this is why your test function is failing

**Comment by zbpetersbuf** [comment]
the function will be printing numbers and returning values but in the test function you are comparing it to just your retuned variables and not the print statement

**Comment by Priyanka-43** [comment]
@zbpetersbuf Before that also it was failing, i was trying different things to see if this somehow get resolved. But if you say i will remove those lines. Because i am not able to find mistake.

**Comment by Priyanka-43** [comment]
I have removed those lines, still showing error.

**Comment by zbpetersbuf** [comment]
@Priyanka-43 ok so now the error is Expected 2, but got 63.6619775813128 this is good 
this means that the function is outputting an expected type of data, its just that the numbers don't match, this is possible due to either the function or test function giving an incorrect output, try different numbers or test functions
perhaps in your module have print statements (you will have to delete these before the pull request is accepted) to see what's going wrong on your end
my honest advise is try @laserlab office
if not that its then to copy your function you are testing and your test function aswell as your pytest issue (found after pressing the blue details on the right) into chat got and see if that helps you

**Comment by laserlab** [comment]
> @Priyanka-43 ok so now the error is Expected 2, but got 63.6619775813128 this is good this means that the function is outputting an expected type of data, its just that the numbers don't match, this is possible due to either the function or test function giving an incorrect output, try different numbers or test functions perhaps in your module have print statements (you will have to delete these before the pull request is accepted) to see what's going wrong on your end my honest advise is try @laserlab office if not that its then to copy your function you are testing and your test function aswell as your pytest issue (found after pressing the blue details on the right) into chat got and see if that helps you

Good case why unit testing is useful. The computer does what you tell it to do and that is you return scipy.simpson of the array you give it. If you read scipy documentation you will read:

```python
Parameters
    ----------
    y : array_like
        Array to be integrated.
    x : array_like, optional
        If given, the points at which `y` is sampled.
    dx : float, optional
        Spacing of integration points along axis of `x`. Only used when
        `x` is None. Default is 1.
    axis : int, optional
        Axis along which to integrate. Default is the last axis.
```
Which gives you the answer what is wrong.

**Comment by Priyanka-43** [comment]
@zbpetersbuf Hi, I have made the required changes. All pylint and pytest are working. If you can merge this pull request.

**Comment by zbpetersbuf** [comment]
@Priyanka-43 you must still resolve conflicts before I am able to merge it

**Comment by Priyanka-43** [comment]
@zbpetersbuf I have done that.

**Pull request merged**